### PR TITLE
protondrive: implement two-password mode

### PR DESCRIPTION
**Pull request by henrybear327**
#### What is the purpose of this change?

Implements the [two-password mode](https://proton.me/support/switch-two-password-mode).

#### Was the change discussed in an issue or in the forum before?

Feature request (or it's kind of a bug :D) from https://github.com/rclone/rclone/issues/6072#issuecomment-1704380975 and https://github.com/rclone/rclone/issues/6072#issuecomment-1704378529.

#### Checklist

- [x] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-new-feature-or-bug-fix).
- [x] I have added tests for all changes in this PR if appropriate.
- [x] I have added documentation for the changes if appropriate.
- [x] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [x] I'm done, this Pull Request is ready for review :-)

### CODE REVIEW

**Review by ncw — COMMENTED** [review]
You haven't set `Advanced` here so this will appear in the normal `rclone config` so you should probably update `protondrive.md` to include it in the example run or set `Advanced: true`

Otherwise looks fine!

File: backend/protondrive/protondrive.go
Comment (henrybear327) [reply backend/protondrive/protondrive.go:75]: After second thought, I will move it to Advance, since Proton's main user base is using 1-password mode!

I will add the reference KB URL in the text!

**Comment by henrybear327** [comment]
FYI @ncw, this PR will help complete all the login methods that Proton is supporting. Please also wait for this to be merged if possible before making a new release :) 

Thank you!

**Review by ncw — APPROVED** [review]
LGTM :-) Merge when happy!

**Pull request merged**
